client/my-sites/people/invite-people/index.jsx

```diff
@@ -270,6 +271,18 @@ const InvitePeople = React.createClass( {
 	},
 
 	render() {
+		if ( this.props.site && ! get( this.props, 'site.capabilities.promote_users' ) ) {
+			return (
```

Comment:
I'm not sure that this is needed quite yet. Currently there's only the one conditional that determine whether we render the form or the error. Also, if the goal is to make `render` small, then we should probably factor out the form into something like `renderInviteForm()`, which would add roughly 70 lines of code to this PR.

I'm not completely against that, but if you think that we should, I'd rather do that in another PR.